lamindb/_curate.py

```diff
@@ -265,28 +277,87 @@ def add_new_from(self, key: str, organism: str | None = None, **kwargs):
         Args:
             key: The key referencing the slot in the DataFrame from which to draw terms.
             organism: The organism name.
-            **kwargs: Additional keyword arguments to pass to the registry model.
+            **kwargs: Additional keyword arguments to pass to create new records
         """
         if len(kwargs) > 0 and key == "all":
             raise ValueError("Cannot pass additional arguments to 'all' key!")
         self._kwargs.update({"organism": organism} if organism else {})
         self._update_registry(key, validated_only=False, **self._kwargs, **kwargs)
 
     def add_new_from_columns(self, organism: str | None = None, **kwargs):
-        """Add validated & new column names to its registry.
+        """Deprecated to run by default during init."""
+        warnings.warn(
+            "`.add_new_from_columns()` is deprecated and will be removed in a future version. It's run by default during initialization.",
+            DeprecationWarning,
+            stacklevel=2,
+        )
+        pass
 
-        Args:
-            organism: The organism name.
-            **kwargs: Additional keyword arguments to pass to the registry model.
+    def _replace_synonyms(
+        self, key: str, syn_mapper: dict, values: pd.Series | pd.Index
+    ):
+        # replace the values in df
+        std_values = values.map(lambda x: syn_mapper.get(x, x))
```

Comment:
```suggestion
        std_values = values.map(lambda unstd_val: syn_mapper.get(unstd_val, unstd_val))
```

Makes it clearer that we default to unstandardized values if we don't find a match.